packages/lib/src/artifacts/ZosContract.ts

```diff
@@ -0,0 +1,86 @@
+import BN from 'bignumber.js';
+import ZWeb3 from './ZWeb3';
+import { getSolidityLibNames, hasUnlinkedVariables } from '../utils/Bytecode';
+import { StorageLayoutInfo } from '../validations/Storage';
+import { Contract, TransactionObject } from 'web3-eth-contract';
+import { TransactionReceipt } from 'web3/types';
+import Contracts, { ContractSchema } from './Contracts';
+import _ from 'lodash';
+
+export default class ZosContract {
+
+  // Wrap Web3 Contract's interface.
+  private _contract: Contract;
+  public get options(): any { return this._contract.options; }
+  public get methods(): any { return this._contract.methods; }
+  public get _address(): string { return this._contract._address; }
+  public async getPastEvents(event: string, options?: any, cb?: any): Promise<any> {
+    return this._contract.getPastEvents(event, options, (error, res) => {
+      if(cb) cb(error, res);
+    });
+  }
+
+  // Extends Web3's Contract interface.
+  public schema: ContractSchema;
+  public binary: string; // TODO: remove
+  public deployedBinary: string; // TODO: remove
+  public storageInfo: StorageLayoutInfo;
+  public warnings: any;
+  public deploymentTransactionReceipt: TransactionReceipt;
+  public deploymentTransactionHash: string;
```

Comment:
Can we store all this under a separate object? I mean, web3 contract has all methods under "methods", events under "events", etc; it'd be nice to bundle all the extra info we need under one or two objects that contain all of them, to avoid polluting the contract interface.

Reply:
The latest version of ZosContract stores schema related stuff (storageInfo and warnings) under ZosContract.schema and deployment related stuff in a `zosData` object, injected into the Web3 Contract object.